assignments/prototypes.js

```diff
@@ -32,7 +62,77 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
- 
+function Humanoid(attr){
+  CharacterStats.call(this,attr);
+  this.team = attr.team;
+  this.weapons = attr.weapons;
+  this.language = attr.language;
+}
+
+//-------------------------------------------Inheritance 
+
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+//-------------------------------------------Prototypes
+
+Humanoid.prototype.greet = function(){
+  return `${this.name} offers a greeting in ${this.language}`;
+}
+
+
+
+
+
+
+ //----------------------------------------------------------------------------------------------------------------
+ //---------------------------------------stretch------------------------------------------------------------------
+
+function Villan(attr){
+  Humanoid.call(this,attr);
+}
+//-------------------------------------------Inheritance
+
+Villan.prototype = Object.create(Humanoid.prototype);
+
+//-------------------------------------------Prototypes
+Villan.prototype.punch = function(){
+  return `${this.name} punched`;
+}
+
+//----------------------------------------------------------------------------------------------------------------
+function Hero(attr){
+  Humanoid.call(this,attr);
+}
+//-------------------------------------------Inheritance 
+
+Hero.prototype = Object.create(Humanoid.prototype);
+
+//-------------------------------------------Prototypes
+
+Hero.prototype.protect = function(){
+  return `the punch was deflected`;
+}
+
+
+//========================================
+let superman = new Hero({
+  name: 'superman',
+  healthPoints: 20
+});
+let batman = new Villan({
+  name: 'batman',
+  health: 20
+});
+ //----------------------------------------------------------------------------------------------------------------
+//-----------------------------------------------------------------------------------------------------------------
+
+
+//console.log(superman);
+console.log(`superman has ${superman.healthPoints} left`);
+console.log(batman.punch());
+console.log(superman.newHealth());
```

Comment:
Great work using a new method to display the new health! 
However, try to think how you can display the new health without using this method!
That's a challenge!